pymc3/stats.py

```diff
@@ -964,3 +964,33 @@ def bfmi(trace):
     energy = trace['energy']
 
     return np.square(np.diff(energy)).mean() / np.var(energy)
+
+
+def r2_score(y_true, y_pred):
+    R"""R-squared for Bayesian regression models. Only valid for linear models.
+    http://www.stat.columbia.edu/%7Egelman/research/unpublished/bayes_R2.pdf
+
+    Parameters
+    ----------
+    y_true: : array-like of shape = (n_samples) or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+    y_pred : array-like of shape = (n_samples) or (n_samples, n_outputs)
+        Estimated target values.
+
+    Returns
+    -------
+    `namedtuple` with the following elements:
+    R2_median: median of the Bayesian R2
+    R2_mean: mean of the Bayesian R2
+    R2_std: standard deviation of the Bayesian R2
+    """
+    e = y_true - y_pred
+    var_y_est = np.var(y_pred, 1)
+    var_e = np.var(e, 1)
+
+    r2_median = np.median(var_y_est / (var_y_est + var_e))
```

Comment:
maybe you should assign `var_y_est / (var_y_est + var_e)` to a variable and then use that variable to compute the mean, median and std